html/includes/table/fdb-search.inc.php

```diff
@@ -35,16 +35,26 @@
 
 if (isset($vars['searchPhrase']) && !empty($vars['searchPhrase'])) {
     $search = mres(trim($vars['searchPhrase']));
-    $ip_search = '%'.mres(trim($vars['searchPhrase'])).'%';
     $mac_search = '%'.str_replace(array(':', ' ', '-', '.', '0x'), '', $search).'%';
 
     if (isset($vars['searchby']) && $vars['searchby'] == 'vlan') {
         $where  .= ' AND `V`.`vlan_vlan` = ?';
         $param[] = (int)$search;
     } elseif (isset($vars['searchby']) && $vars['searchby'] == 'ip') {
+        $ip = $vars['searchPhrase'];
+        $ip_search = '%'.mres(trim($ip)).'%';
         $sql .= " LEFT JOIN `ipv4_mac` AS `M` USING (`mac_address`)";
         $where  .= ' AND `M`.`ipv4_address` LIKE ?';
         $param[] = $ip_search;
+    } elseif (isset($vars['searchby']) && $vars['searchby'] == 'dnsname') {
+        $ip = gethostbyname($vars['searchPhrase']);
+        $ip_search = '%'.mres(trim($ip)).'%';
+        $sql .= " LEFT JOIN `ipv4_mac` AS `M` USING (`mac_address`)";
```

Comment:
Search by says 'dnsname', but you are searching the ipv4_mac table.

Reply:
So what would you call it? The dns name is resolved with gethostbyname to get an ip address which we can search for in lilbrenms db. So even if we end up by searching for ip address the search phrase the user inputs is a dns record name.

Reply:
Yeah, I see now I missed that earlier, probably too sleepy.

I do wish you would try to avoid DNS lookups as much as possible though, they are slow. You could use our \LibreNMS\Util\IP class to test if it is an IP or not.

Reply:
Yeah, good point about dns lookups. I try to avoid it as much as possible by only resolving one result (the top result). Avoiding it more would mean scrapping the feature altogether, which I of course could do but I really like it...  :)

Where would you suggest validating the ip would fit?

Reply:
Many people add devices by DNS name.  It makes managing and searching for them easier.  If a histname is already a DNS name, why do a DNS lookup?